api/queue/queue.go

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2023 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package queue
+
+import (
+	"github.com/go-vela/server/router/middleware/claims"
+	"net/http"
+
+	"github.com/go-vela/types/library"
+
+	"github.com/gin-gonic/gin"
+	"github.com/sirupsen/logrus"
+)
+
+// swagger:operation POST /api/v1/queue/queue-registration admin QueueRegistration
+//
+// Get queue credentials
+//
+// ---
+// produces:
+// - application/json
+// security:
+//   - ApiKeyAuth: []
+// responses:
+//   '200':
+//     description: Successfully generated registration token
+//     schema:
+//       "$ref": "#/definitions/Token"
+//   '401':
+//     description: Unauthorized
+//     schema:
+//       "$ref": "#/definitions/Error"
+
+// Registration represents the API handler to
+// provide queue credentials after worker onboarded with registration token.
+func Registration(c *gin.Context) {
```

Comment:
personal preference:
```suggestion
// Register represents the API handler to
// retrieve queue credentials as part of worker onboarding.
func Register(c *gin.Context) {
```